Show initialization failures also in the popover

as-cii opened the pull request:
Fixes #210 

Previously, we would only show a notification when failing to initialize the client (e.g. this could happen when an error occurs while checking the protocol version) without informing the popover. This was also causing exceptions like #210, because we were failing to instantiate objects like `PortalBindingManager` due to the initialization error.

With this pull-request we will:

* Show an error message in the popover too, coloring the status bar indicator as red when that happens.
* Don't attempt to re-initialize the client if it is outdated or an error has previously occurred.

<p align=center>
<img src="https://user-images.githubusercontent.com/482957/32887103-06dfc82a-cac3-11e7-8b25-9b220594cf7b.png">
</p>

I have tested this works correctly with a few themes (One Light UI, One Dark UI, Atom Material, Nord UI) by opening the DevTools and enabling the `Offline` checkbox in the Network tab. On reload, I verified that the above message was being shown in the popover.

/cc: @jasonrudolph @nathansobo

jasonrudolph commented on styles/teletype.less:
```diff
@@ -416,3 +416,9 @@
+  color: @text-color;
+  text-align: center;
+  padding: 0 @component-padding @component-padding @component-padding;
+}
```
Since these are the same styles used for `.PackageOutdatedComponent`, should we get rid of the duplication?

```less
.PackageOutdatedComponent, .PackageInitializationErrorComponent {
  color: @text-color;
  text-align: center;
  padding: 0 @component-padding @component-padding @component-padding;
}
```

as-cii replied:
Good point, I will address that shortly.

as-cii commented:
> Can I talk you into describing the steps you used to test these changes?

Thanks for reminding me of that, I have updated the body of the pull request to describe the steps I followed.

I have also addressed https://github.com/atom/teletype/pull/212#discussion_r151428811 and verified it works as expected, so I am going to merge this once CI is green. Thanks for the review, @jasonrudolph! 🙇
